src/chainstate/stacks/miner.rs

```diff
@@ -1263,7 +1281,18 @@ impl StacksBlockBuilder {
             let num_mblocks = parent_microblocks.len();
             let last_mblock_hdr = parent_microblocks[num_mblocks - 1].header.clone();
             self.set_parent_microblock(&last_mblock_hdr.block_hash(), last_mblock_hdr.sequence);
-        }
+
+            let mut microblock_cost = tx.cost_so_far();
+            microblock_cost
+                .sub(&parent_block_cost)
+                .expect("BUG: block_cost + microblock_cost < block_cost");
+
+            // if we get here, then we need to reset the block-cost back to 0 because this begins the
+            // block defined by this miner.
```

Comment:
lol copypasta

Reply:
I specifically altered "epoch" to "block"! Is it more confusing? I'm not sure. But having two definitions of epochs is terrifying to me!

Reply:
No, not a problem at all.  Just happened to notice that almost-the-same comment is in `append_block()` ;)